src/irmin-pack/irmin_pack.mli

```diff
@@ -56,7 +56,7 @@ module Make_ext
        and type metadata = Metadata.t
        and type Key.step = Path.step
 
-  val integrity_check : Format.formatter -> repo -> unit
+  val integrity_check : ?ppf:Format.formatter -> repo -> unit
```

Comment:
I think this should probably return something other than `unit` now. I see 4 possible output states:

1. integrity check ran successfully, found errors and repaired successfully;
1. integrity check ran successfully, found errors and failed to repair them (exception raised, etc.);
1. integrity check ran successfully, finding no errors;
1. integrity check failed.

It's probably fine to leave (2) and (4) as exceptions, but I think (1) and (3) should be distinguished to allow external metrics to be kept.

There is also the question of whether to make repair optional, which I think is a good idea. Perhaps `?auto_repair:unit option` or similar?

Reply:
Fixed by returning a result with more details instead :heavy_check_mark: